graphql_compiler/schema_generation/orientdb/get_orientdb_schema_graph.py

```diff
@@ -0,0 +1,464 @@
+        class_name = class_definition['name']
+        immediate_superclass_names = get_superclasses_from_class_definition(
+            class_definition)
+
+        inheritance_set = set(immediate_superclass_names)
+        inheritance_set.add(class_name)
+
+        # Since the input data must be in topological order, the superclasses of
+        # the current class should have already been processed.
+        # A KeyError on the following line would mean that the input
+        # was not topologically sorted.
+        inheritance_set.update(chain.from_iterable(
+            inheritance_sets[superclass_name]
+            for superclass_name in immediate_superclass_names
+        ))
+
+        # Freeze the inheritance set so it can't ever be modified again.
+        inheritance_sets[class_name] = frozenset(inheritance_set)
+    return inheritance_sets
+
+
+def get_superclasses_from_class_definition(class_definition):
+    """Extract a list of all superclass names from a class definition dict."""
+    # New-style superclasses definition, supporting multiple-inheritance.
+    superclasses = class_definition.get('superClasses', None)
+
+    if superclasses:
+        return list(superclasses)
+
+    # Old-style superclass definition, single inheritance only.
+    superclass = class_definition.get('superClass', None)
+    if superclass:
+        return [superclass]
+
+    # No superclasses are present.
+    return []
+
+
+def _get_non_graph_elements(class_name_to_definition, inheritance_sets):
+    """Return a dict mapping class name to NonGraphElement."""
+    non_graph_elements = dict()
+    non_graph_class_names = _get_class_names_of_kind(inheritance_sets, Kind.NonGraph)
+
+    for class_name in non_graph_class_names:
+        class_definition = class_name_to_definition[class_name]
+        class_fields = _get_class_fields(class_definition)
+        abstract = _is_abstract(class_definition)
+
+        inherited_property_definitions = _get_inherited_property_definitions(
+            inheritance_sets[class_name], class_name_to_definition)
+        link_property_definitions, non_link_property_definitions = (
+            _get_link_and_non_link_properties(inherited_property_definitions))
+
+        if len(link_property_definitions) > 0:
+            raise AssertionError(u'There are links {} defined on non-edge class {}'
+                                 .format(link_property_definitions, class_name))
+
+        property_name_to_descriptor = _get_element_properties(
+            class_name, non_link_property_definitions, non_graph_class_names)
+
+        non_graph_elements[class_name] = NonGraphElement(
+            class_name, abstract, property_name_to_descriptor, class_fields)
+    return non_graph_elements
+
+
+def _get_edge_elements(class_name_to_definition, inheritance_sets):
+    """Return a dict mapping class name to EdgeType."""
+    edge_elements = dict()
+    subclass_sets = get_subclass_sets_from_inheritance_sets(inheritance_sets)
```

Comment:
I choose to recompute the subclass_sets multiple times throughout the code instead of passing it as an argument to reduce the number of arguments.